component/src/main/java/org/ballerinalang/sap/nativeimpl/producer/SendBAPIMessage.java

```diff
@@ -0,0 +1,130 @@
+        String response = null;
+        try {
+            OMElement omData = AXIOMUtil.stringToOM(content);
+            String functionName = omData.getQName().toString();
+            if (doLogon) {
+                if (!(StringUtils.isNullOrEmpty(productManufacturer) || StringUtils.isNullOrEmpty(productName) ||
+                        StringUtils.isNullOrEmpty(ccmsInterface))) {
+                    log.error("The logon's parameter/s has/have null value.");
+                }
+                log.info("Begin Transaction");
+                JCoContext.begin(destination);
+                TransactionHandler.logon(destination, productManufacturer, productName, ccmsInterface,
+                        cccmsInterfaceVersion, context);
+            }
+            if (doTransaction) {
+                log.info("Begin transaction.");
+                JCoContext.begin(destination);
+                JCoFunction function = TransactionHandler.getRFCfunction(destination, functionName, context);
+                processMetaDataDocument(omData, function);
+                response = TransactionHandler.evaluateRFCfunction(function, destination, context);
+                JCoFunction commitFunction = TransactionHandler.getRFCfunction(destination,
+                        SapConstants.BAPI_COMMIT_NAME, context);
+                TransactionHandler.evaluateRFCfunction(commitFunction, destination, context);
+                context.setReturnValues(new BString(response));
+                log.info("Commit transaction.");
+            } else {
+                JCoFunction function = TransactionHandler.getRFCfunction(destination, functionName, context);
+                RFCMetaDataParser.processMetaDataDocument(omData, function);
+                response = TransactionHandler.evaluateRFCfunction(function, destination, context);
+                context.setReturnValues(new BString(response));
+            }
+        } catch (XMLStreamException e) {
+            context.setReturnValues(createError(context, "Error occurred when converting the XML data" +
+                    " in as string to XML: " + e.toString()));
+        } catch (Exception e) {
+            if (doTransaction) {
+                JCoFunction rollbackFunction = TransactionHandler.getRFCfunction(destination,
+                        SapConstants.ROLLBACK_BAPI_NAME, context);
+                TransactionHandler.evaluateRFCfunction(rollbackFunction, destination, context);
+                log.info("Rollback transaction.");
+            }
+        } finally {
+            if (doTransaction || doLogon) {
+                try {
+                    JCoContext.end(destination);
+                } catch (JCoException e) {
+                    context.setReturnValues(createError(context, "Error occurred when end the " +
+                            "Jco context."));
+                }
+                log.info("End transaction.");
```

Comment:
Should this be an info log. Should we add some value to the log i.e transaction id ?